Obvs.EventStore.Tests/TestServiceBus.cs

```diff
@@ -16,13 +17,14 @@ public class TestServiceBus
         public async void TestServiceBusWithLocalEventStore()
         {
             var serviceBus = ServiceBus.Configure()
-                .WithEventStoreEndpoints<ITestMessage1>()
-                    .Named("Obvs.EventStore.Test")
-                    .AppendMessageProperties(message => null)
-                    .FilterReceivedMessages(properties => true)
-                    .ConnectToEventStore("ConnectTo=tcp://admin:changeit@127.0.0.1:1113")
-                    .SerializedAsJson()
-                    .AsClientAndServer()
+                .WithEventStoreSharedConnectionScope("ConnectTo=tcp://admin:changeit@127.0.0.1:1113", config => config
+                    .WithEventStoreEndpoints<ITestMessage1>()
+                        .Named("Obvs.EventStore.Test")
+                        .AppendMessageProperties(message => null)
+                        .FilterReceivedMessages(properties => true)
+                        .ConnectToEventStore("ConnectTo=tcp://admin:changeit@127.0.0.1:1113")
```

Comment:
Yeh - a runtime check that assert either a shared connection is in scope
(when UsingSharedConnection()), and if not, ConnectToEventStore() is to be
used as normal

On Tue, 12 Apr 2016 at 23:18 Christopher Read <notifications@github.com>
wrote:

> In Obvs.EventStore.Tests/TestServiceBus.cs
> <https://github.com/megakid/Obvs.EventStore/pull/9#discussion_r59464058>:
>
> > @@ -16,13 +17,14 @@ public class TestServiceBus
> >          public async void TestServiceBusWithLocalEventStore()
> >          {
> >              var serviceBus = ServiceBus.Configure()
> > -                .WithEventStoreEndpoints<ITestMessage1>()
> > -                    .Named("Obvs.EventStore.Test")
> > -                    .AppendMessageProperties(message => null)
> > -                    .FilterReceivedMessages(properties => true)
> > -                    .ConnectToEventStore("ConnectTo=tcp://admin:changeit@127.0.0.1:1113")
> > -                    .SerializedAsJson()
> > -                    .AsClientAndServer()
> > +                .WithEventStoreSharedConnectionScope("ConnectTo=tcp://admin:changeit@127.0.0.1:1113", config => config
> > +                    .WithEventStoreEndpoints<ITestMessage1>()
> > +                        .Named("Obvs.EventStore.Test")
> > +                        .AppendMessageProperties(message => null)
> > +                        .FilterReceivedMessages(properties => true)
> > +                        .ConnectToEventStore("ConnectTo=tcp://admin:changeit@127.0.0.1:1113")
>
> I did give it some thought, the options were:
>
>    - Change fluent config to allow ConnectToEventStore() to be optional,
>    and throw if it isn't set
>    - Add another fluent config method called UsingSharedConnection() and
>    throw if that's not set
>    - Duplicate the set of fluent config interfaces completely for use
>    with shared connection config, and remove the ConnectToEventStore() method
>    from the duplicated interfaces.
>
> Thoughts?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/megakid/Obvs.EventStore/pull/9/files/4ee0eb84386592a162b555badcc686f6f635bec4#r59464058>
>